### ISSUE TITLE
Accessible properties AccessKey and KeyboardShortcut ignore UseMnemonic = false

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#9296]
### .NET version

.NET 8.0.0-preview.6.23307.4

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

Button, Label and derived controls have [mnemonics](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.label.usemnemonic) feature, which allows making a shortcut with one of characters in control's Text. For example, a button with `UseMnemonic = true` and `Text = "&Button"` will be displayed as "<ins>B</ins>utton" and can be activated by pressing Alt+B. If `UseMnemonic` is set to `false`, ampersand will be treated as a regular character, and Alt+B shortcut won't work.

Accessible objects set mnemonic shortcut in `AccessKey` and `KeyboardShortcut` properties, which makes screen readers read the shortcut when control is focused. However, when mnemonic is disabled by setting `UseMnemonic = false`, accessible properties ignore it and still set the shortcut. Screen readers will read the shortcut, even though it doesn't work.

#### Actual behavior

https://github.com/dotnet/winforms/assets/102954094/2a100682-d544-4d6a-848e-e25f4ab7a2b7

Narrator reads the form as follows:

> Mnemonic, edit, Alt+ m, 
> No mnemonic, edit, Alt+ n, 
> Mnemonic, button, Alt+ e, 
> No mnemonic, button, Alt+ o, 
> Mnemonic, radio button, selected, Alt+ i, 
> Mnemonic, radio button, non-selected, Alt+ i, 
> No mnemonic, radio button, selected, Alt+ c, 

For controls with "No mnemonic" text, pressing announced shortcut will not do anything

#### Expected behavior

* `AccessKey` and `KeyboardShortcut` should be empty when mnemonics are disabled
* Narrator should not read shortcut when mnemonics are disabled
* Narrator should read ampersand as part of the text when mnemonics are disabled, exactly as displayed

### Steps to reproduce

Run sample app: 
[MnemonicRepro.zip](https://github.com/dotnet/winforms/files/11759886/MnemonicRepro.zip)

Most likely it also reproduces with all controls derived from Button and Label.

**Comment by merriemcgaw** [comment]
Thanks @dmitrii-drobotov we'll get this on a backlog. The default for `UseMnemonic` is `false` right?

**Comment by dmitrii-drobotov** [comment]
@merriemcgaw - no, default value is `true` both for `Label` and `ButtonBase`.

**Comment by merriemcgaw** [comment]
Hah, I meant to say `True`. I must not have had enough coffee yesterday 😆. But that seems right to me, and since the default behavior is OK I'm not going to consider this a panic issue, but hopefully a good first issue.

**Comment by Amy-Li03** [comment]
Verified this on .NET latest build: 8.0.100-rc.1.23369.12, issue was fixed. AccessKey and KeyboardShortcut properties don't exist, Narrator doesn't read shortcut and read ampersand as part of the text when mnemonics are disabled.
![image](https://github.com/dotnet/winforms/assets/62929087/d9f08696-9490-4ce6-9edf-0e23277e843d)
![image](https://github.com/dotnet/winforms/assets/62929087/ac664e6b-7788-4fc0-9c2d-6fbaf9bef196)

**Comment by MelonWang1** [comment]
Verified with .NET SDK 8.0.100-preview.7.23376.3 test pass build, this issue was fixed, test result same as above.
